x-pack/plugins/timelines/server/search_strategy/index_fields/index.ts

```diff
@@ -36,61 +41,113 @@ export const indexFieldsProvider = (): ISearchStrategy<
   const beatFields: BeatFields = require('../../utils/beat_schema/fields').fieldsBeat;
 
   return {
-    search: (request, options, deps) => from(requestIndexFieldSearch(request, deps, beatFields)),
+    search: (request, options, deps) =>
+      from(requestIndexFieldSearch(request, deps, beatFields, getStartServices)),
   };
 };
 
-export const requestIndexFieldSearch = async (
-  request: IndexFieldsStrategyRequest,
-  { esClient }: SearchStrategyDependencies,
-  beatFields: BeatFields
-): Promise<IndexFieldsStrategyResponse> => {
-  const indexPatternsFetcherAsCurrentUser = new IndexPatternsFetcher(esClient.asCurrentUser);
-  const indexPatternsFetcherAsInternalUser = new IndexPatternsFetcher(esClient.asInternalUser);
-
-  const dedupeIndices = dedupeIndexName(request.indices);
-
-  const responsesIndexFields = await Promise.all(
-    dedupeIndices
+export const findExistingIndices = async (
+  indices: string[],
+  esClient: ElasticsearchClient
+): Promise<boolean[]> =>
+  Promise.all(
+    indices
       .map(async (index) => {
-        if (
-          request.onlyCheckIfIndicesExist &&
-          (index.includes(apmIndexPattern) || index.includes(apmDataStreamsPattern))
-        ) {
-          // for apm index pattern check also if there's data https://github.com/elastic/kibana/issues/90661
-          const searchResponse = await esClient.asCurrentUser.search({
+        if ([apmIndexPattern, apmDataStreamsPattern].includes(index)) {
+          const searchResponse = await esClient.search({
```

Comment:
use `search` instead of `fieldCaps` for these patterns for reasons https://github.com/elastic/kibana/issues/90661